### PR TITLE
[ISSUE 6444]Optimize task node helper document in workflow toolbar

### PR DESCRIPTION
**Pull request by baisui1981**
relevant ISSUE:[https://github.com/apache/dolphinscheduler/issues/6444](https://github.com/apache/dolphinscheduler/issues/6444)

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

File: dolphinscheduler-ui/src/js/conf/home/pages/dag/_source/formModel/formModel.vue [hunk 1]
```diff
@@ -550,7 +554,18 @@
         backfillRefresh: true,
         // whether this is a new Task
         isNewCreate: true,
-        tasksTypeList: Object.keys(tasksType)
+        tasksTypeList: Object.keys(tasksType),
+        helpUrlEnable: function (typeKey) {
```
Comment (songjianet) [inline dolphinscheduler-ui/src/js/conf/home/pages/dag/_source/formModel/formModel.vue:558]: typekey => {}

Comment (songjianet) [inline dolphinscheduler-ui/src/js/conf/home/pages/dag/_source/formModel/formModel.vue]: typekey => {}

File: dolphinscheduler-ui/src/js/conf/home/pages/dag/_source/formModel/formModel.vue [hunk 2]
```diff
@@ -550,7 +554,18 @@
...
+          let type = tasksType[typeKey]
```
Comment (songjianet) [inline dolphinscheduler-ui/src/js/conf/home/pages/dag/_source/formModel/formModel.vue]: const type = tasksType[typeKey]

File: dolphinscheduler-ui/src/js/conf/home/pages/dag/_source/formModel/formModel.vue [hunk 3]
```diff
@@ -550,7 +554,18 @@
...
+          if (type) {
+            let disabled = !!type.helperLinkDisable
+            return !disabled
+          }
+          return false
```
Comment (songjianet) [inline dolphinscheduler-ui/src/js/conf/home/pages/dag/_source/formModel/formModel.vue]: if (!type) return false

const disabled = !! type.helperLinkDisable
return !disabled

There are three inversion operations here, can't it be simplified?

File: dolphinscheduler-ui/src/js/conf/home/pages/dag/_source/formModel/formModel.vue [hunk 4]
```diff
@@ -550,7 +554,18 @@
...
+        },
+        helpUrl: function (tskType) {
```
Comment (songjianet) [inline dolphinscheduler-ui/src/js/conf/home/pages/dag/_source/formModel/formModel.vue]: tasktype => {}

Here `tsktype` should be replaced with `tasktype`.

**Review by songjianet — CHANGES_REQUESTED** [review]

File: dolphinscheduler-ui/src/js/conf/home/pages/dag/_source/formModel/formModel.vue (hunk 1 above)
Comment (wangyizhi1) [inline dolphinscheduler-ui/src/js/conf/home/pages/dag/_source/formModel/formModel.vue]: Functions should be written in methods. [methods](https://cn.vuejs.org/v2/api/#methods)

File: dolphinscheduler-ui/src/js/conf/home/pages/dag/_source/formModel/formModel.vue [hunk 5]
```diff
@@ -17,7 +17,10 @@
 <template>
   <div class="form-model-wrapper" v-clickoutside="_handleClose">
     <div class="title-box">
-      <span class="name">{{ $t("Current node settings") }}</span>
+      <span class="name">{{ $t("Current node settings") }}
+        <a v-if="helpUrlEnable(nodeData.taskType)" class="helper-link" target="_blank"
+           :href="helpUrl(nodeData.taskType)">?{{nodeData.taskType}} {{ $t('Instructions') }}</a>
```
Comment (songjianet) [inline dolphinscheduler-ui/src/js/conf/home/pages/dag/_source/formModel/formModel.vue]: It is recommended to change the question mark to an icon. The icon can be set in the same style as the text.
``` vue
<a v-if="helpUrlEnable(nodeData.taskType)" class="helper-link" target="_blank" :href="helpUrl(nodeData.taskType)">
  <i class="el-icon-question" />
  {{nodeData.taskType}} {{ $t('Instructions') }}
</a>
```

**Review by songjianet — CHANGES_REQUESTED** [review]

File: dolphinscheduler-ui/src/js/conf/home/pages/dag/_source/formModel/formModel.vue (hunk 3 above)
Comment (baisui1981) [reply dolphinscheduler-ui/src/js/conf/home/pages/dag/_source/formModel/formModel.vue]: ok , has been modified
``` javascript
helpUrlEnable: function (typeKey) {
          const type = tasksType[typeKey]
          if (type) {
            if (!type.helperLinkDisable) return true
            return !type.helperLinkDisable
          }
          return false
        },
```

File: dolphinscheduler-ui/src/js/conf/home/pages/dag/_source/formModel/formModel.vue (hunk 4 above)
Comment (baisui1981) [reply dolphinscheduler-ui/src/js/conf/home/pages/dag/_source/formModel/formModel.vue]: ok

File: dolphinscheduler-ui/src/js/conf/home/pages/dag/_source/formModel/formModel.vue (hunk 5 above)
Comment (baisui1981) [reply dolphinscheduler-ui/src/js/conf/home/pages/dag/_source/formModel/formModel.vue]: tks, you have provided so beautify icon

File: dolphinscheduler-ui/src/js/conf/home/pages/dag/_source/formModel/formModel.vue (hunk 2 above)
Comment (baisui1981) [reply dolphinscheduler-ui/src/js/conf/home/pages/dag/_source/formModel/formModel.vue]: ok

File: dolphinscheduler-ui/src/js/conf/home/pages/dag/_source/formModel/formModel.vue (hunk 1 above)
Comment (baisui1981) [reply dolphinscheduler-ui/src/js/conf/home/pages/dag/_source/formModel/formModel.vue]: ok

**Review by songjianet — DISMISSED** [review]

**Comment by songjianet** [comment]
LGTM

File: dolphinscheduler-ui/src/js/conf/home/pages/dag/_source/formModel/formModel.vue (hunk 1 above)
Comment (baisui1981) [reply dolphinscheduler-ui/src/js/conf/home/pages/dag/_source/formModel/formModel.vue]: then when i add  **methods** ,vue module throw an exception as blow, is there something invalid for me ?
```
vue.runtime.esm.js?2b0e:1897 TypeError: _vm.helpUrlEnable is not a function
    at Proxy.render (formModel.vue?d1b5:22)
    at VueComponent.Vue._render (vue.runtime.esm.js?2b0e:3569)
    at VueComponent.updateComponent (vue.runtime.esm.js?2b0e:4081)
    at Watcher.get (vue.runtime.esm.js?2b0e:4495)
    at new Watcher (vue.runtime.esm.js?2b0e:4484)
    at mountComponent (vue.runtime.esm.js?2b0e:4088)
    at VueComponent.Vue.$mount (vue.runtime.esm.js?2b0e:8459)
    at init (vue.runtime.esm.js?2b0e:3137)
    at createComponent (vue.runtime.esm.js?2b0e:6022)
    at createElm (vue.runtime.esm.js?2b0e:5969)
```

Comment (baisui1981) [reply dolphinscheduler-ui/src/js/conf/home/pages/dag/_source/formModel/formModel.vue]: ok, i have catch your idea, and have modified it

**Review by wangyizhi1 — APPROVED** [review]
LGTM

**Review by songjianet — APPROVED** [review]

**Comment by songjianet** [comment]
LGTM

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**